diff --git a/linkcheck/url.py b/linkcheck/url.py
index 5da62cd..b920d58 100644
--- a/linkcheck/url.py
+++ b/linkcheck/url.py
@@ -437,7 +437,7 @@ def match_host (host, domainlist):
     return False
 
 
-_nopathquote_chars = "-;/=,~*+()@!"
+_nopathquote_chars = "-;/=,~*+()@!:"
 if os.name == 'nt':
     _nopathquote_chars += "|"
 _safe_url_chars = re.escape(_nopathquote_chars + "_:.&#%?[]!")+"a-zA-Z0-9"
