diff --git a/tools/firmware/Makefile b/tools/firmware/Makefile
index b966c68..61ad1d5 100644
--- a/tools/firmware/Makefile
+++ b/tools/firmware/Makefile
@@ -15,7 +15,7 @@ SUBDIRS-$(CONFIG_OVMF) += ovmf-dir
 SUBDIRS-$(CONFIG_SEABIOS) += seabios-dir
 SUBDIRS-$(CONFIG_ROMBIOS) += rombios
 SUBDIRS-$(CONFIG_ROMBIOS) += vgabios
-SUBDIRS-$(CONFIG_PV_SHIM) += xen-dir
+#SUBDIRS-$(CONFIG_PV_SHIM) += xen-dir
 SUBDIRS-y += hvmloader
 
 LD32BIT-$(CONFIG_FreeBSD) := LD32BIT_FLAG=-melf_i386_fbsd
@@ -54,10 +54,10 @@ endif
 ifeq ($(CONFIG_OVMF),y)
 	$(INSTALL_DATA) ovmf-dir/ovmf.bin $(INST_DIR)/ovmf.bin
 endif
-ifeq ($(CONFIG_PV_SHIM),y)
-	$(INSTALL_DATA) xen-dir/xen-shim $(INST_DIR)/xen-shim
-	$(INSTALL_DATA) xen-dir/xen-shim-syms $(DEBG_DIR)/xen-shim-syms
-endif
+#ifeq ($(CONFIG_PV_SHIM),y)
+#	$(INSTALL_DATA) xen-dir/xen-shim $(INST_DIR)/xen-shim
+#	$(INSTALL_DATA) xen-dir/xen-shim-syms $(DEBG_DIR)/xen-shim-syms
+#endif
 
 .PHONY: uninstall
 uninstall:
@@ -68,10 +68,10 @@ endif
 ifeq ($(CONFIG_OVMF),y)
 	rm -f $(INST_DIR)/ovmf.bin
 endif
-ifeq ($(CONFIG_PV_SHIM),y)
-	rm -f $(INST_DIR)/xen-shim
-	rm -f $(DEBG_DIR)/xen-shim-syms
-endif
+#ifeq ($(CONFIG_PV_SHIM),y)
+#	rm -f $(INST_DIR)/xen-shim
+#	rm -f $(DEBG_DIR)/xen-shim-syms
+#endif
 
 .PHONY: clean
 clean: subdirs-clean
