--- common/macros.c	2012-06-21 14:58:41.000000000 +0400
+++ common/macros.c	2013-02-21 12:24:19.000000000 +0400
@@ -2586,6 +2586,7 @@
 	for (x = 0; x < MACRO_X_COUNT; x++) {
 		macro_keys[x].code = x;
 		macro_keys[x].name = macro_x_names[x];
+		macro_keys[x].clean_options = 0;
 
 		/* host/service output/perfdata and author/comment macros should get cleaned */
 		if((x >= 16 && x <= 19) || (x >= 49 && x <= 52) || (x >= 99 && x <= 100) || (x >= 124 && x <= 127)) {
